### PR TITLE
48 sync actions of the plugin configuration addremove scene with build settings

### LINKED ISSUE #48: Sync actions of the plugin configuration (add/remove scene) with build settings
When we add or remove the scene from the plugin settings - this is an explicit action, so we can add or remove the scene from the build setting and display notification.

When Editor build settings is 100% in sycn we can just clean & rebuild it
When it's out of sync we will just add / remove the scene without carrying for the order.

### PR DESCRIPTION
**Pull request by ghost**
## Purpose of this PR
When Editor build settings is 100% in sync we just clean & rebuild scenes from Buid Settings

## Testing status
* No tests have been added.

## Comments to reviewers
Depends on "#53" PR

### CODE REVIEW

File: com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsConfig.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        internal static bool ShowOutOfSyncPreventingDialog
+        {
+            get => s_ShowOutOfSyncPreventingDialog;
+            set
+            {
+                s_ShowOutOfSyncPreventingDialog = value;
+                PlayerPrefs.SetInt(k_ShowOutOfSyncPreventingDialogPref, value ? 1 : 0);
+                PlayerPrefs.Save();
```
Comment (stan-osipov) [inline com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsConfig.cs:25]: why are we using player prefs here? What if the user will clean it?

File: com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsValidator.cs [hunk 2]
```diff
@@ -13,9 +13,44 @@ public class BuildConfigurationSettingsValidator
...
+            if (!outOfSync)
+            {
+                return;
+            }
+            
+            var result = EditorUtility.DisplayDialogComplex(
+                "Scenes Management",
+                EditorBuildSettingsValidator.ScenesSyncWarningDescription,
+                "Ok, continue",
+                "Cancel",
```
Comment (stan-osipov) [inline com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsValidator.cs:40]: Cancel, exit plamode

File: com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsValidator.cs [hunk 3]
```diff
@@ -13,9 +13,44 @@ public class BuildConfigurationSettingsValidator
...
+                "Ok, neven show again");
```
Comment (stan-osipov) [inline com.stansassets.scene-management/BuildConfigurator/Editor/BuildConfigurationSettingsValidator.cs:41]: Don't show again

**Comment by stan-osipov** [comment]
This Pr has the wrong branch target and makes it very hard to review.

File: com.stansassets.scene-management/BuildConfigurator/Editor/EditorNBuildSettingsDrawer.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
+using System.Linq;
+using StansAssets.Plugins.Editor;
+using UnityEditor;
+using UnityEngine;
+
+namespace StansAssets.SceneManagement.Build
+{
+    internal class EditorNBuildSettingsDrawer
```
Comment (stan-osipov) [inline com.stansassets.scene-management/BuildConfigurator/Editor/EditorNBuildSettingsDrawer.cs:8]: you can skip internal

File: com.stansassets.scene-management/BuildConfigurator/Editor/EditorNBuildSettingsDrawer.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        }
+
+        struct AutoSyncParams
+        {
+            public int LastScenesCount;
+            public bool Synced;
+            public bool NeedScenesSync;
+
+            public AutoSyncParams(int lastScenesCount, bool synced, bool needScenesSync)
```
Comment (stan-osipov) [inline com.stansassets.scene-management/BuildConfigurator/Editor/EditorNBuildSettingsDrawer.cs:117]: looks like this is unsused.

**Review by stan-osipov — CHANGES_REQUESTED** [review]
Some questions and the biggest problem so far is that it wasn't working for me. whe I add scene to the scene manager I do not see those in the build configuration.

**Comment by yurii-lunha-stansassets** [comment]
I'll replace this PR with similar.

**Pull request closed without merging**